sphinx/ext/doctest.py

```diff
@@ -101,12 +126,32 @@ def run(self):
             # parse doctest-like output comparison flags
             option_strings = self.options['options'].replace(',', ' ').split()
             for option in option_strings:
-                if (option[0] not in '+-' or option[1:] not in
-                        doctest.OPTIONFLAGS_BY_NAME):  # type: ignore
-                    # XXX warn?
+                on_or_off, option_name = option[0], option[1:]
+                if on_or_off not in '+-':  # type: ignore
+                    self.state.document.reporter.warning(
+                        "missing '+' or '-' in '%s' option." % option,
```

Comment:
Could you wrap with `_(...)` function?
Ex:
```
from sphinx.locale import _
self.state.document.reporter.warning(_("missing '+' or '-' in '%s' option.") % option)
```

Then it makes the message translatable.